### ISSUE TITLE
Make gesture focal point configurable

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#4924]
**tldr we need to expose a user configurable focal point. A focal represents the x,y value on which a gesture works. In normal cases the x,y from the MotionEvent is used. We need to make this configurable**

Currently we already expose an API that works similar to a  focal point implementation. When you enable the tracking mode on MyLocationView you also change the focal point to follow that view. An example how that now looks:

```java
                    // Single finger double tap
                    if (mMapboxMap.getTrackingSettings().isLocationTrackingDisabled()) {
                        // Zoom in on gesture
                        zoom(true, e.getX(), e.getY());
                    } else {
                        // Zoom in on user location view
                        // getCenterX also takes in account padding + insets
                        zoom(true, mMyLocationView.getCenterX(), mMyLocationView.getCenterY());
                    }
```

To be able to support all different use-cases users are requesting we need to expose an API as:
 
```java
mapboxMap.getUiSettings().setFocalPoint(x,y);
```

People will mostly use this to center the focal point 

```java
mapboxMap.getUiSettings().setFocalPoint(mapView.getWidth()/2,mapView.getHeight()2));
```

It would be nice to be able to limit if-else in the gesture code above and use a factory pattern to provide an implementation instead.

**Comment by 1ec5** [comment]
A common use case would be to focus zooming on the selected marker. In fact, I suspect that would account for the majority of requests for this feature.

**Comment by 1ec5** [comment]
For the iOS SDK, a configurable gesture anchor point was introduced in #5302. It affects all gestures except for panning.

**Comment by tobrun** [comment]
Here you can see the gesture in action:

![ezgif com-video-to-gif 32](https://cloud.githubusercontent.com/assets/2151639/16007321/0936afa8-3172-11e6-8543-6785c6b383f8.gif)

**Comment by bleege** [comment]
Squashed, Rebased, and Merged into `release-android-v4.1.0`.
